### PR TITLE
Support Elasticsearch 7.x

### PR DESCRIPTION
**Pull request by berggren**
## One line description of pull request
Support Elasticsearch 7.x

## Description:
Support Elasticsearch 7.x but stay backwards compatible with versions prior to 7.
Tested with ES 7 and 6 server versions

**Related issue (if applicable):** fixes #<plaso issue number here>
#2465

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/l2tdocs/blob/master/process/Code%20review%20process.asciidoc). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://plaso.readthedocs.io/en/latest/sources/developer/Style-guide.html).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
  - [ ] Automated checks (Travis, Codecov, Codefactor )pass
  - [ ] No new [new dependencies](https://plaso.readthedocs.io/en/latest/sources/developer/Adding-a-new-dependency.html) are required or l2tdevtools has been updated
  - [ ] Reviewer assigned

### CODE REVIEW

**Comment by joachimmetz** [comment]
Please fix linting

```
************* Module plaso.output.shared_elastic

plaso/output/shared_elastic.py:114:0: C0330: Wrong hanging indentation (add 2 spaces).

        body=self._event_documents, index=self._index_name,

        ^ | (bad-continuation)

plaso/output/shared_elastic.py:115:0: C0330: Wrong hanging indentation (add 2 spaces).

        request_timeout=self._DEFAULT_REQUEST_TIMEOUT)

        ^ | (bad-continuation)
```

File: plaso/output/shared_elastic.py [hunk 1]
```diff
@@ -110,9 +110,15 @@ def _FlushEvents(self):
       # pylint does not recognizes request_timeout as a valid kwarg. According
       # to http://elasticsearch-py.readthedocs.io/en/master/api.html#timeout
       # it should be supported.
-      self._client.bulk(
-          body=self._event_documents, doc_type=self._document_type,
-          index=self._index_name, request_timeout=self._DEFAULT_REQUEST_TIMEOUT)
+      bulk_args = dict(
+        body=self._event_documents, index=self._index_name,
```
Comment (joachimmetz) [inline plaso/output/shared_elastic.py]: `+2` continuation indentation

File: plaso/output/shared_elastic.py [hunk 2]
```diff
@@ -110,9 +110,15 @@ def _FlushEvents(self):
       # pylint does not recognizes request_timeout as a valid kwarg. According
       # to http://elasticsearch-py.readthedocs.io/en/master/api.html#timeout
       # it should be supported.
-      self._client.bulk(
-          body=self._event_documents, doc_type=self._document_type,
-          index=self._index_name, request_timeout=self._DEFAULT_REQUEST_TIMEOUT)
+      bulk_args = dict(
```
Comment (joachimmetz) [inline plaso/output/shared_elastic.py]: maybe use `{}` instead of dict?

File: plaso/output/shared_elastic.py [hunk 3]
```diff
@@ -110,9 +110,15 @@ def _FlushEvents(self):
...
+        body=self._event_documents, index=self._index_name,
+        request_timeout=self._DEFAULT_REQUEST_TIMEOUT)
+
+      # TODO: Remove once Elasticsearch v6.x is deprecated.
+      if self.GetClientMajorVersion() < 7:
```
Comment (joachimmetz) [inline plaso/output/shared_elastic.py]: I opt to store the major version in a variable instead of calling GetClientMajorVersion multiple times

Comment (joachimmetz) [reply plaso/output/shared_elastic.py]: nvm

File: plaso/output/shared_elastic.py [hunk 4]
```diff
@@ -248,6 +258,15 @@ def Close(self):
 
     self._client = None
 
+  def GetClientMajorVersion(self):
```
Comment (joachimmetz) [inline plaso/output/shared_elastic.py]: Since this is only used internally please prefix with an underscore

File: plaso/output/shared_elastic.py [hunk 5]
```diff
@@ -248,6 +258,15 @@ def Close(self):
...
+    """Get the major version of the Elasticsearch client library.
+
+    Returns:
+      int: Major version number.
+
```
Comment (joachimmetz) [inline plaso/output/shared_elastic.py]: remove while line

File: plaso/output/shared_elastic.py [hunk 6]
```diff
@@ -110,9 +110,15 @@ def _FlushEvents(self):
       # pylint does not recognizes request_timeout as a valid kwarg. According
```
Comment (joachimmetz) [inline plaso/output/shared_elastic.py]: It looks like this comment might be no longer relevant, remove?

File: plaso/output/shared_elastic.py (hunk 4 above)
Comment (berggren) [reply plaso/output/shared_elastic.py]: done

File: plaso/output/shared_elastic.py (hunk 5 above)
Comment (berggren) [reply plaso/output/shared_elastic.py]: Done

File: plaso/output/shared_elastic.py (hunk 6 above)
Comment (berggren) [reply plaso/output/shared_elastic.py]: Done

File: plaso/output/shared_elastic.py (hunk 2 above)
Comment (berggren) [reply plaso/output/shared_elastic.py]: Done

File: plaso/output/shared_elastic.py (hunk 1 above)
Comment (berggren) [reply plaso/output/shared_elastic.py]: Done

**Comment by berggren** [comment]
@joachimmetz PTAL

**Pull request merged**